docs/content/how-to/kubevirt/ingress-and-dns.md

```diff
@@ -0,0 +1,213 @@
+
+In lieu of the default ingress and DNS behavior, it is also possible to
+configure a Hypershift KubeVirt guest cluster with a unique base domain
+at creation time. This option does require some manual configuration
+steps during creation though.
+
+This process involves three steps:
+
+1. Cluster creation
+2. LoadBalancer creation
+3. Wildcard DNS configuration
+
+### Step 1 - Deploying the HostedCluster specifying our base domain
+
+```shell linenums="1"
+export CLUSTER_NAME=example
+export PULL_SECRET="$HOME/pull-secret"
+export MEM="6Gi"
+export CPU="2"
+export WORKER_COUNT="2"
+export BASE_DOMAIN=hypershift.lab
+
+hypershift create cluster kubevirt \
+--name $CLUSTER_NAME \
+--node-pool-replicas $WORKER_COUNT \
+--pull-secret $PULL_SECRET \
+--memory $MEM \
+--cores $CPU \
+--base-domain $BASE_DOMAIN
+```
+
+With above configuration we will end up having a HostedCluster with an ingress wildcard configured for `*.apps.example.hypershift.lab` (*.apps.<hostedcluster_name\>.<base_domain\>).
+
+This time, the HostedCluster will not finish the deployment (will remain in `Partial` progress) as we saw in the previous section, since we have configured a base domain we need to make sure that the required DNS records and load balancer are in-place:
+
+```shell linenums="1"
+oc get --namespace clusters hostedclusters
+
+NAME            VERSION   KUBECONFIG                       PROGRESS   AVAILABLE   PROGRESSING   MESSAGE
+example                   example-admin-kubeconfig         Partial    True        False         The hosted control plane is available
+```
+
+If we access the HostedCluster this is what we will see:
+
+```shell
+hypershift create kubeconfig --name $CLUSTER_NAME > $CLUSTER_NAME-kubeconfig
+```
+
+```shell
+oc --kubeconfig $CLUSTER_NAME-kubeconfig get co
+
+NAME                                       VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+console                                    4.12.7    False       False         False      30m     RouteHealthAvailable: failed to GET route (https://console-openshift-console.apps.example.hypershift.lab): Get "https://console-openshift-console.apps.example.hypershift.lab": dial tcp: lookup console-openshift-console.apps.example.hypershift.lab on 172.31.0.10:53: no such host
+.
+.
+.
+ingress                                    4.12.7    True        False         True       28m     The "default" ingress controller reports Degraded=True: DegradedConditions: One or more other status conditions indicate a degraded state: CanaryChecksSucceeding=False (CanaryChecksRepetitiveFailures: Canary route checks for the default ingress controller are failing)
+```
+
+In the next section we will fix that.
+
+### Step 2 - Set up the LoadBalancer
+
+
+!!! note
+
+    If your cluster is on bare-metal you may need MetalLB to be able to provision functional LoadBalancer services. Take a look at the section [Optional MetalLB Configuration Steps](#optional-metallb-configuration-steps).
```

Comment:
[Optional MetalLB Configuration Steps ] link does not seem to be working here.

Reply:
the same link seems to work when the docs are published. maybe they don't work when viewing in github as markdown though. 

https://hypershift-docs.netlify.app/how-to/kubevirt/create-kubevirt-cluster/#optional-metallb-configuration-steps